pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_events.go

```diff
@@ -325,51 +314,42 @@ func (d duplicateEventsEvaluator) testDuplicatedEvents(testName string, flakeOnl
 				continue
 			}
 		}
-		eventDisplayMessage, times := GetTimesAnEventHappened(fmt.Sprintf("%s - %s", event.Locator, event.Message))
+
+		times := GetTimesAnEventHappened(event.StructuredMessage)
 		if times > DuplicateEventThreshold {
-			// If we marked this message earlier in recordAddOrUpdateEvent as interesting/true, we know it matched one of
-			// the existing patterns or one of the AllowedRepeatedEventFns functions returned true.
-			if strings.Contains(eventDisplayMessage, InterestingMark) {
+
+			// Check if we have an allowance for this event. This code used to just check if it had an interesting flag,
+			// implying it matches some pattern, but that happens even for upgrade patterns occurring in non-upgrade jobs,
+			// so we were ignoring patterns that were meant to be allowed only in upgrade jobs in all jobs. The list of
+			// allowed patterns passed to this object wasn't even used.
+			if allowed, _ := MatchesAny(d.allowedDupeEvents, event.StructuredLocator, event.StructuredMessage, kubeClientConfig, &d.topology); allowed {
 				continue
 			}
 
-			eventMessageString := eventDisplayMessage + " From: " + event.From.Format("15:04:05Z") + " To: " + event.To.Format("15:04:05Z")
-			if _, ok := displayToCount[eventMessageString]; !ok {
```

Comment:
Pretty sure this was busted, we're using eventMessageString which contains timestamps, not eventDisplayMessage, so our attempt to put unique occurences of a message into a map key was probably broken. I think you can see this in the repeating events listed when the test fails.